pkgs/applications/editors/lapce/utils.nix

```
@@ -0,0 +1,80 @@
+{
+  stdenvNoCC,
+  curl,
+  cacert,
+  zstd,
+  linkFarm,
+}:
+let
+  fetchPluginTarballFromRegistry =
+    {
+      author,
+      name,
+      version,
+      hash,
+      meta,
+    }:
+    stdenvNoCC.mkDerivation (
+      let
+        url = "https://plugins.lapce.dev/api/v1/plugins/${author}/${name}/${version}/download";
+        file = "lapce-plugin-${author}-${name}-${version}.tar.zstd";
+      in
+      {
+        name = file;
+        nativeBuildInputs = [
+          curl
+          cacert
+        ];
+        phases = [ "installPhase" ];
+        installPhase = ''
+          url="$(curl ${url})"
+          curl -L "$url" -o "$out"
+        '';
```

Add missing install hooks (same for 2nd one)
```suggestion
        installPhase = ''
          runHook preInstall

          url="$(curl ${url})"
          curl -L "$url" -o "$out"

          runHook postInstall
        '';
```